src/api/modules/sign/chunks.ts

```diff
@@ -33,41 +35,64 @@ const chunks: {
  * @returns Response from the background
  */
 export const sendChunk = (chunk: Chunk) =>
-  new Promise<void>((resolve, reject) => {
+  new Promise<void>(async (resolve, reject) => {
     const callID = nanoid();
+
     // construct message
     const message: ApiCall = {
+      app:
+        import.meta.env?.VITE_IS_EMBEDDED_APP === "1"
+          ? "wanderEmbedded"
+          : "wander",
+      // TODO: Add Wallet API version:
+      version: "",
+      callID,
       type: "chunk",
-      app: "wander",
-      data: chunk,
-      callID
+      data: chunk
     };
 
-    // send message
-    window.postMessage(message, window.location.origin);
+    if (import.meta.env?.VITE_IS_EMBEDDED_APP === "1") {
```

Comment:
This can be implemented better by hiding this conditional inside `isomorphicSendMessage`. I'll create a refactoring task for later.